### PR TITLE
Add tags for buffer-local variants of each linter option

### PR DESCRIPTION
**Pull request by adriaanzon**
I blindly went ahead and added buffer-local variables for each linter option to the documentation. I also added an example to use the buffer-local variables with shellcheck, but I noticed that the shellcheck linter doesn't use the `ale#Var` function. This may also be the case in other linters, so consider this pull request a WIP while I'm going to check that.

I also made some stylistic changes.

### CODE REVIEW

**Comment by adriaanzon** [comment]
Shouldn't `ale_linters_sh_shellcheck_exclusions` be called `ale_sh_shellcheck_exclusions`?

File: doc/ale-handlebars.txt [hunk 1]
```diff
@@ -6,9 +6,8 @@ ALE Handlebars Integration                             *ale-handlebars-options*
 ember-template-lint                          *ale-handlebars-embertemplatelint*
 
 g:ale_handlebars_embertemplatelint_executable
-                              \ *g:ale_handlebars_embertemplatelint_executable*
-
-  Type: |String|
+                                *g:ale_handlebars_embertemplatelint_executable*
+  Type: |String|                  *b:ale_handlebars_embertemplatelint_executable*
```
Comment (adriaanzon) [inline doc/ale-handlebars.txt]: This one was a bit weird to format but I've done it like in other places where there is a tag for `g:` and `b:` variables: One line between the variable name and the beginning of Type / Default.

Like this one: https://github.com/adriaanzon/ale/blob/7230cbe9e43d64a02212ae4803abff874e39b8d6/doc/ale.txt#L366

**Comment by w0rp** [comment]
Yes, `ale_linters_sh_shellcheck_exclusions` should be called `ale_sh_shellcheck_exclusions`. We'll have to add some tests for that and set up a fallback. Then we can remove the code for the old option completely in 2.0.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**